services/src/main/java/edu/unc/lib/dl/cdr/services/rest/modify/AddContainerController.java

```diff
@@ -15,67 +15,88 @@
-            @RequestParam(value = "type", required = false) String type,
-            @RequestParam(value = "description", required = false) MultipartFile description, Model model,
-            HttpServletRequest request, HttpServletResponse response) {
-        PID parent = new PID(parentId);
-//        try {
-//            String user = request.getHeader("On-Behalf-Of");
-//            if (user == null) {
-//                user = request.getRemoteUser();
-//            }
-//
-//            ContentModelHelper.Model extraModel = null;
-//            if (type != null && type.equals("collection")) {
-//                extraModel = ContentModelHelper.Model.COLLECTION;
-//            } else if (type != null && type.equals("aggregate")) {
-//                extraModel = ContentModelHelper.Model.AGGREGATE_WORK;
-//            }
-//
-//            byte[] mods = null;
-//            if (description != null && !description.isEmpty()) {
-//                mods = description.getBytes();
-//            }
-//
-//            PID containerPid = digitalObjectManager.createContainer(name, parent, extraModel, user, mods);
-//            response.setStatus(201);
-//            return "{\"pid\": \"" + containerPid.getPid() + "\"}";
-//        } catch (IOException e) {
-//            log.error("Unexpected IO exception", e);
-//        } catch (IngestException e) {
-//            log.error("Unexpected exception", e);
-//        }
-        response.setStatus(500);
-        return "{\"error\": \"An error occurred while attempting to create a container in " + parent + "\"}";
+    @Autowired
+    private AddContainerService addContainerService;
+
+    @RequestMapping(value = "edit/create/adminUnit/{id}", method = RequestMethod.POST)
+    @ResponseBody
+    public ResponseEntity<Object> createAdminUnit(@PathVariable("id") String id) {
+        return createContainer(id, Cdr.AdminUnit);
+    }
+
+    @RequestMapping(value = "edit/create/collection/{id}", method = RequestMethod.POST)
+    @ResponseBody
+    public ResponseEntity<Object> createCollection(@PathVariable("id") String id) {
+        return createContainer(id, Cdr.Collection);
+    }
+
+    @RequestMapping(value = "edit/create/folder/{id}", method = RequestMethod.POST)
+    @ResponseBody
+    public ResponseEntity<Object> createFolder(@PathVariable("id") String id) {
+        return createContainer(id, Cdr.Folder);
+    }
+
+    @RequestMapping(value = "edit/create/work/{id}", method = RequestMethod.POST)
+    @ResponseBody
+    public ResponseEntity<Object> createWork(@PathVariable("id") String id) {
+        return createContainer(id, Cdr.Work);
+    }
+
+    private ResponseEntity<Object> createContainer(String id, Resource containerType) {
+        Map<String, Object> result = new HashMap<>();
+        result.put("action", "create");
+
+        PID parentPid = PIDs.get(id);
+        result.put("pid", parentPid.getUUID());
```

Comment:
This is fine, since if we move to an asynchronous model later we may not know the pid of the newly created container.  At some point we will probably want to revisit the responses from these if we want to improve responsiveness of the admin ui